extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcSession.java

```diff
@@ -18,10 +19,29 @@ public interface OidcSession {
     /**
      * Return an {@linkplain:Instant} indicating how long will it take for the current session to expire.
      *
-     * @return
+     * @deprecated Use {@link #expiresAt()}.
+     *
+     * @return Instant
      */
+    @Deprecated
     Instant expiresIn();
 
+    /**
+     * Return an {@linkplain Instant} representing the current session's expiration time
+     * which is a number of seconds from the epoch of 1970-01-01T0:0:0Z.
```

Comment:
Is this javadoc correct? It looks like the javadoc of `expiresIn`? I thought we decided to display the actual expiration time here?
`which is a number of seconds from the epoch of 1970-01-01T0:0:0Z` looks weird.

Reply:
@gsmet It is correct, this is what the `exp` claim value is which is what this `Instant` captures

Reply:
I don't understand really. Isn't this method supposed to return the real expiration time and not something weird? The doc seems to indicate it will be 1970 + 30 minutes (typically) instead of the real expiration time?

I thought it was the original issue we were trying to solve so I might miss something entirely.

Reply:
@gsmet It is just the way the expiration time is handled for JWT tokens, the epoch is a base. if we take from some other base then we won't be able to correctly compare it against the current time (which is also calculated from the epoch).
From the `exp` claim definition in the OIDC spec:
```
Expiration time on or after which the ID Token MUST NOT be accepted for processing. ... Its value is a JSON number representing the number of seconds from 1970-01-01T0:0:0Z".
The same for `issuedAt` time, etc.

> The doc seems to indicate it will be 1970 + 30 minutes
It will be more than 30 mins, it will be more than the current time. If it is less than the current time then the session (token) has expired.

Reply:
@gsmet, if you'd like you can do `mvn quarkus:dev` in `security-openid-connect-quickstart` and go to the OIDC devUI card, login to KC as `alice:alice` and check the `exp` value in the displayed access or ID token...

Reply:
Removed the ref to the epoch time to avoid the confusion as agreed with Guillaume